### ISSUE TITLE
GDScript compiler: Leaving block with non-zero temporary variables

### ISSUE THREAD

**Issue by jivvy** [godotengine/godot#62549]
### Godot version

v4.0.alpha.custom_build [622b656c4]

### System information

Arch Linux, Sway/Wayland

### Issue description

Error message in console:
```
ERROR: Leaving block with non-zero temporary variables: 1
   at: pop_stack_identifiers (modules/gdscript/gdscript_byte_codegen.h:151)
ERROR: Non-zero temporary variables at end of function: 1
   at: write_end (modules/gdscript/gdscript_byte_codegen.cpp:180)
```

### Steps to reproduce

1. New project
2. Add a Node2D. I called mine 'World'
3. Attach script to 'World'
4. Add this function:
```gdscript
func set_pos(pos: Vector2, obj: Object):
	var ma = []
	ma[pos.x] = []
	ma[pos.x][pos.y] = obj
```

**Comment by cdemirer** [comment]
Already fixed as part of #62485.

**Comment by jivvy** [comment]
My mistake. Wasn't running the newly built Godot binaries correctly.
